sql/logic_test.go

```diff
@@ -436,7 +437,25 @@ func (t *logicTest) execQuery(query logicQuery) {
 			t.Fatalf("%s: expected %s, but found %s", query.pos, query.expectedHash, hash)
 		}
 	} else if !reflect.DeepEqual(query.expectedResults, results) {
-		t.Fatalf("%s: expected %q, but found %q\n", query.pos, query.expectedResults, results)
+		var buf bytes.Buffer
+		w := tabwriter.NewWriter(&buf, 2, 1, 2, ' ', 0)
+
+		fmt.Fprintf(w, "%s: expected:\n", query.pos)
+		for i := 0; i < len(query.expectedResults); i += len(cols) {
+			for _, value := range query.expectedResults[i : i+len(cols)] {
+				fmt.Fprintf(w, "%q\t", value)
+			}
+			fmt.Fprint(w, "\n")
+		}
+		fmt.Fprint(w, "but found:\n")
+		for i := 0; i < len(results); i += len(cols) {
+			for _, value := range results[i : i+len(cols)] {
+				fmt.Fprintf(w, "%q\t", value)
+			}
+			fmt.Fprint(w, "\n")
+		}
+		w.Flush()
+		t.Fatal(buf.String())
```

Comment:
It's nice that this is a separate commit, but really this should be a separate PR as it is unrelated to the conversion of the tests.